pkg/reconciler/intevents/topic/topic.go

```diff
@@ -156,8 +160,16 @@ func (r *Reconciler) reconcileTopic(ctx context.Context, topic *v1.Topic) error
 			logging.FromContext(ctx).Desugar().Error("Topic does not exist and the topic policy doesn't allow creation")
 			return fmt.Errorf("Topic %q does not exist and the topic policy doesn't allow creation", topic.Spec.Topic)
 		} else {
+			topicConfig := &pubsub.TopicConfig{}
+			if r.dataresidencyStore != nil {
+				if dataresidencyCfg := r.dataresidencyStore.Load(); dataresidencyCfg != nil {
+					if dataresidencyCfg.DataResidencyDefaults.ComputeAllowedPersistenceRegions(topicConfig) {
+						r.Logger.Infow("Updated Topic Config", zap.Any("topicConfig", *topicConfig))
+					}
+				}
+			}
```

Comment:
Is there a unit test for this code? I couldn't find one.

Reply:
This code still uses gpubsub. Instead of adding tests here, @zhongduo and I agreed that we should try to switch away from gpubsub (#1739) first and follow up with tests after.